suzieq/engines/pandas/interfaces.py

```diff
@@ -89,6 +87,69 @@ def get(self, **kwargs):
         else:
             return df.reset_index(drop=True)[fields]
 
+    def _check_vlan_match(self, vlan_list: List[str],
+                          df: pd.DataFrame) -> pd.DataFrame:
+        '''Return a dataframe with rows in VLANs requested
+
+        VLAN is treated specially because its the only field that is a list
+        of integers, and can be filtered on multiple fields: vlan and vlanList
+        A user filter on VLAN is a list that can contain numeric comparisons
+        as well as simple integers.
+
+        Its not made into a generic filter match because this match on VLAN
+        applies only to interface table.
+        '''
+
+        # If there are any VLANs with ! mixed without !, remove the
+        # ! ones. vlanList is an OR list, not an AND list, except
+        # if all are !
+        if any(x.startswith('!') for x in vlan_list):
+            cond = 'and'
+        else:
+            cond = 'or'
+
+        resdf_list = []
+        if (any(x.startswith('>') for x in vlan_list) and
+                any(x.startswith('<') for x in vlan_list)):
+            # if the user specifies both >/>= and </<=, then we
+            # treat this as an and
+            cond = 'and'
+
+        opdict = {'<': operator.lt, '>': operator.gt,
```

Comment:
We defined this dictionary in many places. Can't we just define a function in utils doing this mapping everywhere? (or I think also storing the dictionary once in a shared location is fine)

Reply:
I thought we had, but I didn't find it. Will look again.

Reply:
Its only in device.py and here. Will see if I can move them into engineobj